Sources/ApolloCodegenLib/FileGenerators/FileGenerator.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+import ApolloUtils
+
+/// The methods to conform to when building a code generation Swift file generator.
+protocol FileGenerator: Equatable {
+  associatedtype graphQLType
+
+  var objectType: GraphQLObjectType { get }
```

Comment:
What is the reason we need to even have this variable on the protocol? Making this a generic protocol feels like it overcomplicates things. Where are we using this `objectType` property?

Reply:
My initial response was that maybe it would make more sense when the other file generators are introduced but looking at that more I think I'm still subconsciously still planning for concurrency where all file generators have a common interface so the concurrency coordinator doesn't need to care what file generator it actually is. So when you remove concurrency then the entire need for this protocol falls away.